orttraining/orttraining/training_ops/cuda/communication/nccl_service.cc

```diff
@@ -0,0 +1,346 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#ifdef USE_NCCL
+
+#include "core/common/common.h"
+#include "core/profile/context.h"
+#include "core/providers/cuda/cuda_common.h"
+#include "orttraining/core/framework/mpi_context.h"
+#include "orttraining/training_ops/cuda/communication/nccl_service.h"
+#include <iostream>
+#include <nccl.h>
+
+namespace onnxruntime {
+namespace cuda {
+
+bool NcclTask::Compare(const NcclTask& other) const {
+  if (type != other.type) {
+    return false;
+  }
+  if (peers.size() != other.peers.size()) {
+    return false;
+  }
+  for (size_t i = 0; i < peers.size(); ++i) {
+    if (peers[i] != other.peers[i]) {
+      return false;
+    }
+  }
+
+  return true;
+}
+
+void NcclTask::ResetTask() {
+  ptr = nullptr;
+  size = 0;
+  is_enqueued = false;
+  is_finished = false;
+}
+
+void NcclTaskGroup::PlanTask(
+    const NcclTask::Type type,
+    const std::vector<int> peers) {
+  batch.push_back({type, peers, nullptr, 0, false, false, ""});
+};
+
+const NcclTask* NcclTaskGroup::EqueueTask(
```

Comment:
>NcclTaskGroup::EqueueTask [](start = 16, length = 25)

I guess I don't fully understand this function... The name indicating it's enqueuing a task, but this function doesn't perform any queuing operation and returns a task pointer..?